src/build.d

```diff
@@ -1390,35 +1347,37 @@ class DependencyRef
 struct MethodInitializer(T)
 {
     private T obj;
-    auto ref opDispatch(string name, T)(T arg)
+    auto ref opDispatch(string name)(typeof(__traits(getMember, T, name)) arg)
     {
         mixin("obj." ~ name ~ " = arg;");
         return this;
     }
 }
 
+/** Create an object using a chain of method calls for each field. */
+T methodInit(T, alias Func, Args...)(Args args)
+{
+    auto initializer = MethodInitializer!T(new T());
+    Func(initializer, initializer.obj, args);
+    initializer.obj.finalize();
```

Comment:
I think in most cases apps would want a `finalize` method if they were using `methodInit`.  The problem is that if the app uses the wrong name or mispells `finalize` then this code will silently not call the method, whereas currently, they'll get an error message that `finalize` doesn't exist.  The drawback is that they need to add an empty `finalize` method but I think that's ok.  Also, if the object is in another library, they may not be able to add a finalize method so it would have to be a UFCS call, meaning you would need to do something like `__traits(is(initializer.obj.finalize()))` or `__traits(compiles, initializer.obj.finalize())`, in which case, if there is an error in the `finalize` function then it would also silently get ignored.